src/systems/filecoin_mining/storage_mining/storage_miner_actor.go

```diff
@@ -90,12 +90,12 @@ func (a *StorageMinerActorCode_I) _isChallenged(rt Runtime) bool {
 	return ret
 }
 
-func (cs *ChallengeStatus_I) ShouldChallenge(currEpoch block.ChainEpoch, minChallengePeriod block.ChainEpoch) bool {
-	return currEpoch > (cs.LastChallengeEpoch()+minChallengePeriod) && !cs.IsChallenged()
+func (cs *ChallengeStatus_I) ShouldChallenge(currEpoch block.ChainEpoch) bool {
+	return currEpoch > (cs.LastChallengeEpoch()+PROVING_PERIOD/2) && !cs.IsChallenged()
 }
 
-func (st *StorageMinerActorState_I) _shouldChallenge(rt Runtime, minChallengePeriod block.ChainEpoch) bool {
```

Comment:
why undo this? it makes more sense to pass it in as a param so the logic can be run entirely from sampleMiners